### ISSUE TITLE
Model with Custom Attribute without PropertyName throws error

### ISSUE THREAD

**Issue by kylednoland** [aliencube/AzureFunctions.Extensions#101]
## Repro Steps

Add an `OpenApiResponseWithBody` on a Function with a response with type that is decorated with a custom attribute without the `PropertyName` field populated.

Example of Function Attribute
```
 [OpenApiResponseWithBody(HttpStatusCode.OK, "application/json", typeof(Thing), Summary = "A returned Thing")]
```

Example of referenced class that produces the error
```
    public class Thing
    {
        [JsonProperty(Required = Required.Default)]
        public string Text { get; set; }
    }
```

Currently, the Function will throw an error when generating an OpenAPI document with a response: `System.Private.CoreLib: Exception while executing function: RenderSwaggerDocument. System.Private.CoreLib: Value cannot be null. Parameter name: key.`.

## Potential Solution

In `PropertyInfoExtensions.cs` add a `null` check for when there is no `PropertyName` and return the `element.Name` property instead such as:

```
        public static string GetJsonPropertyName(this PropertyInfo element)
        {
            if (element.ExistsCustomAttribute<JsonPropertyAttribute>())
            {
                var name = element.GetCustomAttribute<JsonPropertyAttribute>().PropertyName;

                if (name == null)
                {
                    return element.Name;
                }

                return name;

            }

            return element.Name;
        }
```

**Comment by justinyoo** [comment]
@kylednoland Thanks for raising the issue! Can you try `[JsonRequired]` for your case, instead of `[JsonProperty(Required = Required.Default)]`

Even if it doesn't solve the issue with the current version, it's sorted out on [v3](https://github.com/aliencube/AzureFunctions.Extensions/tree/v3), which will be released soon. Please have a try and let us know.

**Comment by kylednoland** [comment]
Hey @justinyoo, thanks for the quick response! I switched out the `JsonProperty` for `JsonRequired` and it seems to have resolved the issue using my example from above. With v3 coming soon would it be worth it to fix this to maintain v2?

**Comment by justinyoo** [comment]
@kylednoland This has been fixed on [v3.0.0](https://www.nuget.org/packages/Aliencube.AzureFunctions.Extensions.OpenApi/3.0.0)

**Comment by kylednoland** [comment]
Thanks for the v3 updates @justinyoo! I just updated my code with the latest NuGet package update for version 3.0.0 and I'm still unfortunately seeing this same error. Updating models to use `[JsonRequired]` rather than `[JsonProperty(Required = Required.Default)]` does seem to still fix the issue (as in v2).

**Comment by kylednoland** [comment]
Looking at the updated `PropertyInfoExtensions.cs` file it looks like it would need to check if the `PropertyName` value is null in addition to the check if the custom property exists in order to resolve this issue.

**Comment by justinyoo** [comment]
@kylednoland Re-open the issue to have a look. Could you please provide me with steps to repro?

I have a sample model like this:

https://github.com/aliencube/AzureFunctions.Extensions/blob/c53adaff65b33bacf7780272e3e36aaf87c2cf6f/samples/Aliencube.AzureFunctions.FunctionApp.Models/DummyResponseModel.cs#L41-L42

The `Required.Default` enum value doesn't mean the property is `required`, by the way. 🤔 It only sees `Required.Always` and/or `Required.DisallowNull` to recognise as `required`.

**Comment by kylednoland** [comment]
To reproduce you would need a model like the following:

```
[JsonProperty(Required = Required.DisallowNull)]
public string JsonPropertyUnNamedValue { get; set; } = "hello world";
```

**Comment by kylednoland** [comment]
Issue fixed in PR #103
